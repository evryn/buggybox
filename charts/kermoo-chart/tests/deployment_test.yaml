suite: test deployment
templates:
  - deployment.yaml
tests:
  -
    it: manifest with default values should match snapshot
    asserts:
      - matchSnapshot: {}
  -
    it: works with extra common labels
    set:
      commonLabels:
        my-label: hello
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: metadata.labels["my-label"]
          value: hello
      - equal:
          path: spec.template.metadata.labels["my-label"]
          value: hello
  -
    it: works with extra common annotations
    set:
      commonAnnotations:
        my-annotation: hello
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: metadata.annotations["my-annotation"]
          value: hello
      - equal:
          path: spec.template.metadata.annotations["my-annotation"]
          value: hello
  -
    it: works with custom image
    set:
      image.registry: custom-registry.io
      image.repository: custom-repo
      image.tag: custom-tag
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].image
          value: custom-registry.io/custom-repo:custom-tag
  -
    it: works with debug enabled
    set:
      debug: true
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].command
          value:
            - start
            - -v
            - debug
  -
    it: works with different replicaCount
    set:
      replicaCount: 99
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.replicas
          value: 99
  -
    it: works with additional container template
    set:
      additionalContainerTemplate:
        resources:
          limits:
            cpu: 250m
            memory: 256Mi
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].resources.limits.cpu
          value: 250m
      - equal:
          path: spec.template.spec.containers[0].resources.limits.memory
          value: 256Mi
  -
    it: works with no services
    set:
      services:
    asserts:
      - isKind:
          of: Deployment
      - notExists:
          path: spec.template.spec.containers[0].ports
  -
    it: works with multiple services
    set:
      services:
        - port: 80
          targetPort: 81
        - port: 8080
          targetPort: 8181
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].ports[0].containerPort
          value: 81
      - equal:
          path: spec.template.spec.containers[0].ports[1].containerPort
          value: 8181