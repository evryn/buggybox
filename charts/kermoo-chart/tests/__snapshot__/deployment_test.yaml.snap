manifest with default values should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-deployment
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: RELEASE-NAME
      template:
        metadata:
          labels:
            app: RELEASE-NAME
        spec:
          containers:
            - command:
                - kermoo
                - start
              image: docker.io/evryn/kermoo:latest
              name: kermoo
              ports:
                - containerPort: 80
                - containerPort: 8080
              volumeMounts:
                - mountPath: ~/.kermoo/config.yaml
                  name: config-volume
                  subPath: config.yaml
          volumes:
            - configMap:
                name: RELEASE-NAME-config
              name: config-volume
